internal/controllers/cluster/cluster_controller_phases.go

```diff
@@ -210,17 +212,26 @@ func (r *Reconciler) reconcileInfrastructure(ctx context.Context, s *scope) (ctr
 	}
 	s.infraCluster = obj
 
-	// Determine if the infrastructure provider is ready.
-	ready, err := external.IsReady(s.infraCluster)
+	// Determine contract version used by the InfrastructureCluster.
+	contractVersion, err := utilconversion.GetContractVersion(ctx, r.Client, obj.GroupVersionKind())
 	if err != nil {
 		return ctrl.Result{}, err
 	}
-	if ready && !cluster.Status.InfrastructureReady {
-		log.Info("Infrastructure provider has completed cluster infrastructure provisioning and reports status.ready", cluster.Spec.InfrastructureRef.Kind, klog.KObj(s.infraCluster))
+
+	// Determine if the InfrastructureCluster is provisioned.
+	provisioned, err := contract.InfrastructureCluster().Provisioned(contractVersion).Get(obj)
+	if err != nil {
```

Comment:
Wondering if we should do it here similar to below so we don't have a pointer to use below.

```suggestion
	var provisioned bool
	if provisionedPtr, err := contract.InfrastructureCluster().Provisioned(contractVersion).Get(obj); err == nil {
		provisioned = *provisionedPtr
	else if err != nil {
```

But also okay to keep as is.

Reply:
Fixed here and also in the Machine controller for consistency